lib/state/ui/reducer.ts

```diff
@@ -90,9 +90,22 @@ const dialogs: A.Reducer<T.DialogType[]> = (state = [], action) => {
   switch (action.type) {
     case 'CLOSE_DIALOG':
       return state.slice(0, -1);
-
+    case 'TRASH_TAG':
+      return state.filter(
+        (element) => element.type !== 'TRASH-TAG-CONFIRMATION'
+      );
+    case 'REMOTE_TAG_DELETE':
+      return state.filter(
+        (element) =>
+          !(
+            element.type === 'TRASH-TAG-CONFIRMATION' &&
+            tagHashOf(element.tagName) === action.tagHash
+          )
+      );
     case 'SHOW_DIALOG':
-      return state.includes(action.dialog) ? state : [...state, action.dialog];
+      return state.find((element) => element.type === action.name)
```

Comment:
this prevents overlapping multiple tag trash dialogs over each other, which I believe is the right move anyway - just noting it. might be worth a comment in the code to tell future us that it's not a mistake

Reply:
Good call. 
I do have a question about this section @dmsnell 

`return state.find((dialog) => dialog.type === action.name)
        ? state
        : [...state, { type: action.name, tagName: action.tagName }];`

Is there some way I can make this more generic? I tried a few things but couldn't get it to work.
If we add more dialogs that have extra data it would have to be manually added here. 
I figure there should be a way to do it because in the `ShowDialog` [type](https://github.com/Automattic/simplenote-electron/blob/add/delete-tag-confirmation-middleware/lib/state/action-types.ts#L108) we are using the generic `data` property. But I can't seem to figure out how to get this to work.

Reply:
I don't know how much more generic we want it. that's a question at least this PR doesn't raise: the generic question would likely be "can we have more than one dialog of a specific type open if they have different data?" for tag deletion it seems reasonable to not allow that, and we don't currently have any other dialogs with associated data.

were you thinking of another kind of generalization?